src/video/kmsdrm/SDL_kmsdrmvideo.c

```diff
@@ -383,16 +383,25 @@ KMSDRM_FBInfo *KMSDRM_FBFromBO(SDL_VideoDevice *_this, struct gbm_bo *bo)
     num_planes = KMSDRM_gbm_bo_get_plane_count(bo);
     for (int i = 0; i < num_planes; i++) {
         strides[i] = KMSDRM_gbm_bo_get_stride_for_plane(bo, i);
-        handles[i] = KMSDRM_gbm_bo_get_handle(bo).u32;
+        handles[i] = KMSDRM_gbm_bo_get_handle_for_plane(bo, i).u32;
         offsets[i] = KMSDRM_gbm_bo_get_offset(bo, i);
         modifiers[i] = modifiers[0];
     }
 
-    if (modifiers[0]) {
+    if (modifiers[0] && modifiers[0] != DRM_FORMAT_MOD_INVALID) {
         flags = DRM_MODE_FB_MODIFIERS;
     }
 
     ret = KMSDRM_drmModeAddFB2WithModifiers(viddata->drm_fd, w, h, format, handles, strides, offsets, modifiers, &fb_info->fb_id, flags);
+
+    if (ret) {
+        SDL_LogDebug(SDL_LOG_CATEGORY_VIDEO, "Failed to use modifiers");
+        SDL_memcpy(handles, (uint32_t [4]){KMSDRM_gbm_bo_get_handle(bo).u32,0,0,0}, 16);
```

Comment:
Can you please expand this out instead of inlining the get_handle call? The same with get_stride, below?

Reply:
To clarify, SDL builds with C89ish compilers, so we shouldn't use more modern language features.

Reply:
should be fine now